src/data-type.js

```diff
@@ -174,24 +198,30 @@ const TYPE = module.exports.TYPE = {
     },
 
     writeParameterData: function(buffer, parameter, options) {
-      if (parameter.value != null) {
-        let days, dstDiff, minutes;
-        if (options.useUTC) {
-          days = Math.floor((parameter.value.getTime() - UTC_EPOCH_DATE.getTime()) / (1000 * 60 * 60 * 24));
-          minutes = (parameter.value.getUTCHours() * 60) + parameter.value.getUTCMinutes();
-        } else {
-          dstDiff = -(parameter.value.getTimezoneOffset() - EPOCH_DATE.getTimezoneOffset()) * 60 * 1000;
-          days = Math.floor((parameter.value.getTime() - EPOCH_DATE.getTime() + dstDiff) / (1000 * 60 * 60 * 24));
-          minutes = (parameter.value.getHours() * 60) + parameter.value.getMinutes();
-        }
+      const value = parameter.value;
+
+      if (value === null || value === undefined) {
+        buffer.writeUInt8(0);
+        return;
+      }
 
-        buffer.writeUInt8(4);
-        buffer.writeUInt16LE(days);
+      if (!(value instanceof Date)) {
+        throw new TypeError(`parameter.value must be a Date, undefined or null. Received type ${typeof value}`);
```

Comment:
This test should also be made in `validate()`, because `Request.validateParameters()` calls `validate()` and generates a better error message if the exception occurs there.
If the error occurs later in the constructor of `RpcRequestPayload`, the error message would not indicate the parameter name.